core/corehttp/gateway_handler.go

```diff
@@ -273,116 +275,82 @@ func (i *gatewayHandler) postHandler(w http.ResponseWriter, r *http.Request) {
 	http.Redirect(w, r, ipfsPathPrefix+k.String(), http.StatusCreated)
 }
 
-func (i *gatewayHandler) putEmptyDirHandler(w http.ResponseWriter, r *http.Request) {
-	newnode := uio.NewEmptyDirectory()
-
-	key, err := i.node.DAG.Add(newnode)
+func (i *gatewayHandler) putHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
should derive a ctx here and cancel it to ensure everything is wiped at the end.

```go
ctx, cancel := context.WithCancel(i.node.Context())
defer cancel()
```